app/javascript/components/history/History.js

```diff
@@ -1,31 +1,134 @@
 import React from 'react';
 import { PropTypes } from 'prop-types';
-import { Card } from 'react-bootstrap';
+import { Button, Card } from 'react-bootstrap';
+import axios from 'axios';
+import reportError from '../util/ReportError';
+import moment from 'moment-timezone';
 
 import { formatTimestamp } from '../../utils/DateTime';
 import moment from 'moment-timezone';
 import { time_ago_in_words } from './helpers';
 
-const history = ({ history }) => {
-  return (
-    <Card className="card-square mt-4 mx-3 shadow-sm">
-      <Card.Header>
-        <b>{history.created_by}</b>, {time_ago_in_words(moment(history.created_at).toDate())} ago ({formatTimestamp(history.created_at)})
-        <span className="float-right">
-          <div className="badge-padding h5">
-            <span className="badge badge-secondary">{history.history_type}</span>
-          </div>
-        </span>
-      </Card.Header>
-      <Card.Body>
-        <Card.Text>{history.comment}</Card.Text>
-      </Card.Body>
-    </Card>
-  );
-};
+class History extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      edit_mode: false,
+      loading: false,
+      comment: '',
+    };
+  }
+
+  handleEditComment = () => {
+    this.setState({ edit_mode: true });
+  };
+
+  handleTextChange = event => {
+    this.setState({ comment: event.target.value });
+  };
+
+  handleDeleteComment = () => {
+    this.setState({ loading: true }, () => {
+      axios.defaults.headers.common['X-CSRF-Token'] = this.props.authenticity_token;
+      axios
+        .delete(window.BASE_PATH + '/histories/' + this.props.history.id)
+        .then(() => {
+          location.reload(true);
+        })
+        .catch(error => {
+          reportError(error);
+        });
+    });
+  };
+
+  formatEditable = () => {
+    if (this.props.history.history_type == 'Comment') {
+      return (
+        <React.Fragment>
+          <Button variant="link" id={this.props.history.id} className="py-0" onClick={this.handleEditComment}>
+            Edit
+          </Button>
+          <Button variant="link" id={this.props.history.id} className="py-0" onClick={this.handleDeleteComment}>
+            Delete
+          </Button>
+        </React.Fragment>
+      );
+    } else {
+      return '';
+    }
+  };
+
+  editComment = () => {
+    this.setState({ loading: true }, () => {
+      axios.defaults.headers.common['X-CSRF-Token'] = this.props.authenticity_token;
+      axios
+        .patch(window.BASE_PATH + '/histories/' + this.props.history.id, {
+          comment: this.state.comment,
+        })
+        .then(() => {
+          location.reload(true);
+        })
+        .catch(error => {
+          reportError(error);
+        });
+    });
+  };
+
+  renderEditMode = () => {
```

Comment:
This is confusing because this renders both edit mode and not, I would remove the conditional and only have this render the edit stuff.  Then add a CardBody in the render below with a conditional that renders either `Card.Text>{this.props.history.comment}</Card.Text>` or `renderEdit`

you also can use `this.state.edit_mode` in the conditional